lib/qif/writer.rb

```diff
@@ -61,11 +62,17 @@ def initialize(io, type = 'Bank', format = 'dd/mm/yyyy')
     
     # Add a transaction for writing
     def <<(transaction)
-      @transactions << transaction
+      case transaction.class.to_s
```

Comment:
I think this part of the PR needs some work. If there can only be one account then it seems it should not be set using the add transaction method here. Nor should it use an array instance variable.